app/sir_trevor_blocks/recent_items_block.rb

```diff
@@ -2,7 +2,9 @@
 
 class RecentItemsBlock < SirTrevorRails::Blocks::SolrDocumentsBlock
   def documents
-    search_results.last
+    search_results.last.select do |doc|
+      doc.has? Spotlight::Engine.config.thumbnail_field
```

Comment:
Did you try implementing this as part of the query instead, so it always returns 8 items if possible? If the most recent 8 happen to have no thumbnail, this won't work even though it could.